### PR TITLE
Add Transaction class

### PR DESCRIPTION
**Pull request by alex-stone**
### What changed? Why?
This adds the Coinbase::Transaction class that will be used by trading, staking, and transfers.

We may want to consider the `sign` function to actually update the `signed_payload` at some point

#### Qualified Impact

This class is currently unused so should have no impact

### CODE REVIEW

File: lib/coinbase/transaction.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      TERMINAL_STATES = [COMPLETE, FAILED].freeze
+    end
+
+    # Returns a new Transaction object. Do not use this method directly.
+    # @param model [Coinbase::Client::Transaction] The underlying Transaction object
+    def initialize(model, from_address_id:)
+      raise unless model.is_a?(Coinbase::Client::Transaction)
+
+      @model = model
+      @from_address_id = from_address_id
```
Comment (alex-stone) [inline lib/coinbase/transaction.rb:38]: Can get this off model now

**Review by erdimaden — APPROVED** [review]

File: lib/coinbase/transaction.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+      # The Transaction is complete and has confirmed on the Network.
+      COMPLETE = 'complete'
+
+      # The Transaction has failed for some reason.
+      FAILED = 'failed'
+
+      # The states that are considered terminal on-chain.
+      TERMINAL_STATES = [COMPLETE, FAILED].freeze
+    end
```
Comment (John-peterson-coinbase) [inline lib/coinbase/transaction.rb:30]: use symbols like in transfer.rb? Ref: https://github.com/coinbase/coinbase-sdk-ruby/blob/5040d0d7a000c24f95f234bd8a0d47f2090b6cb5/lib/coinbase/transfer.rb#L18

**Review by John-peterson-coinbase — APPROVED** [review]

File: lib/coinbase/transaction.rb (hunk 2 above)
Comment (alex-stone) [reply lib/coinbase/transaction.rb:30]: We are removing Transfer::Status in the next PR.

We are returning strings from the API and I think we should just be using the frozen strings in this case

Comment (alex-stone) [reply lib/coinbase/transaction.rb:30]: All of our statuses are returned via strings and all comparisons should use these constants and not the symbol.

I think using the frozen string for comparison purposes is cleanest here!

**Pull request merged**